packages/core/saved-objects/core-saved-objects-api-server-internal/src/lib/repository.ts

```diff
@@ -351,28 +353,34 @@ export class SavedObjectsRepository implements ISavedObjectsRepository {
-    const spacesToAuthorize = new Set(existingNamespaces);
-    spacesToAuthorize.delete(ALL_NAMESPACES_STRING); // Don't accidentally check for global privileges when the object exists in '*'
-    const authorizationResult = await this._securityExtension?.performAuthorization({
-      // If a user tries to create an object with `initialNamespaces: ['*']`, they need to have 'create' privileges for the Global Resource
-      // (e.g., All privileges for All Spaces).
-      // Inversely, if a user tries to overwrite an object that already exists in '*', they don't need to 'create' privileges for the Global
-      // Resource, so in that case we have to filter out that string from spacesToAuthorize (because `allowGlobalResource: true` is used
-      // below.)
-      actions: new Set(['create']),
-      types: new Set([type]),
-      spaces: new Set([...spacesToEnforce, ...spacesToAuthorize]), // existing namespaces are included so we can later redact if necessary
-      enforceMap: new Map([[type, spacesToEnforce]]),
-      auditCallback: (error) =>
-        this._securityExtension!.addAuditEvent({
-          action: AuditAction.CREATE,
-          savedObject: { type, id },
-          error,
-          ...(!error && { outcome: 'unknown' }), // If authorization was a success, the outcome is unknown because the create operation has not occurred yet
-        }),
-      options: { allowGlobalResource: true },
+    // const spacesToEnforce = new Set(initialNamespaces).add(namespaceString); // Always check/enforce authZ for the active space
+    // const existingNamespaces = preflightResult?.existingDocument?._source?.namespaces || [];
+    // const spacesToAuthorize = new Set(existingNamespaces);
+    // spacesToAuthorize.delete(ALL_NAMESPACES_STRING); // Don't accidentally check for global privileges when the object exists in '*'
+    // const authorizationResult = await this._securityExtension?.performAuthorization({
+    //   // If a user tries to create an object with `initialNamespaces: ['*']`, they need to have 'create' privileges for the Global Resource
+    //   // (e.g., All privileges for All Spaces).
+    //   // Inversely, if a user tries to overwrite an object that already exists in '*', they don't need to 'create' privileges for the Global
+    //   // Resource, so in that case we have to filter out that string from spacesToAuthorize (because `allowGlobalResource: true` is used
+    //   // below.)
+    //   actions: new Set([SecurityAction.CREATE]),
+    //   types: new Set([type]),
+    //   spaces: new Set([...spacesToEnforce, ...spacesToAuthorize]), // existing namespaces are included so we can later redact if necessary
+    //   enforceMap: new Map([[type, spacesToEnforce]]),
+    //   options: { allowGlobalResource: true },
+    //   auditOptions: { objects: [{ type, id }] },
+    // });
+
+    const authorizationResult = await this._securityExtension?.authorizeCreate({
```

Comment:
The above commented code is replaced by this extension call, which handles figuring out how to authorize the create action based on the the preflight results.

Reply:
I think we need the bulk vs non-bulk for traceability. See [comment above](https://github.com/elastic/kibana/pull/148165#discussion_r1061603985). Do you prefer having two exposed methods, or an options flag?

Reply:
> Do you prefer having two exposed methods, or an options flag

The first thing that comes to my mind is to have two separate methods, but I don't have a strong preference, so feel free to pick the approach you like better.

Reply:
I implemented separate public methods that call a universal internal method. I think this is best option for external usability. Thanks, Oleg!